nomad/structs/structs.go

```diff
@@ -1172,6 +1172,7 @@ func (n *Node) TerminalStatus() bool {
 // Stub returns a summarized version of the node
 func (n *Node) Stub() *NodeListStub {
 	return &NodeListStub{
+		Address:           n.Attributes["unique.network.ip-address"],
```

Comment:
Oops, a coworker pointed out this should use the _advertise_ address for the node. Unfortunately we only have the HTTP IP+Port available on the Node struct, so you'll have to use net.SplitHostPort on it:

```go
addr, _, _ := net.SplitHostPort(n.HTTPAddr)

return &NodeListStub{
	Address:           addr,
	...
```